public/js/eventsPage.js

```diff
@@ -92,18 +94,53 @@ function displayEventsHandler(events) {
     }
 };
 
-async function displayEvents(events) {
+async function displayEvents(events, hiringPartners) {
     events.forEach(async (event) => {
-        eventGenerator(event)
+        eventGenerator(event, hiringPartners)
+        .then((event) => {
+            let data = {
+                event_id: event.id
+            }
+
+            let hiringPartnersDiv = document.querySelector(`.hiring-partners[data-eventId='${event.id}']`)
+
+            fetch('./api/getHpsByEventId', {
+                credentials: 'same-origin',
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json',
+                },
+                method: 'post',
+                body: JSON.stringify(data)
+
+        })
+        .then(response => response.json())
+        .then(response => {
+            if(response.length != 0) {
+                hiringPartnersDiv.innerHTML += `<h4>Attending hiring partners</h4>`
```

Comment:
these should be aggregated into a large string before being output all at once - at the moment if an error happens half way through then only part of the data will be output